src/babashka/bbin/scripts.clj

```diff
@@ -292,32 +293,58 @@
                                      {:nofollow-links true})]
     (install-script script-file script-contents (:dry-run cli-opts))))
 
+
+(defn main-ns [manifest-resource]
+  (with-open [is (io/input-stream manifest-resource)]
+    (some-> (java.util.jar.Manifest. is)
+                (.getMainAttributes)
+                ;; FIXME .getValue is missing in Babashka? Create issue. See https://docs.oracle.com/javase/7/docs/api/java/util/jar/Attributes.html
+                #_(.getValue "Main-Class")
+                (->> (into {}) (some (fn [[k v]]
+                                       (when (clojure.string/includes? k "Main-Class") v))))
+                (clojure.main/demunge))))
+
+
+(defn jar->main-ns [jar-path]
+  (let [jar-file (io/file jar-path)
+        manifest-path "META-INF/MANIFEST.MF"]
+    (or (with-open [jar (java.util.jar.JarFile. jar-file)]
+          (when (.getEntry jar manifest-path)
+            (main-ns (java.net.URL. "jar" nil (str "file:" (.getAbsolutePath jar-file) "!/" manifest-path)))))
+
+        (throw (ex-info "jar has no Main-Class" {:babashka/exit 1})))))
```

Comment:
The current implementation is causing some [test failures in Windows](https://github.com/babashka/bbin/actions/runs/3891827078/jobs/6721062643). I was able to fix this by simplifying to the following code:
```clojure
(defn jar->main-ns [jar-path]
  (with-open [jar-file (JarFile. (fs/file jar-path))]
    (or (some-> (.getManifest jar-file)
                (.getMainAttributes)
                ;; FIXME .getValue is missing in Babashka? Create issue.
                ;; See https://docs.oracle.com/javase/7/docs/api/java/util/jar/Attributes.html
                #_(.getValue "Main-Class")
                (->> (into {}) (some (fn [[k v]]
                                       (when (clojure.string/includes? k "Main-Class") v))))
                (clojure.main/demunge))
        (throw (ex-info "jar has no Main-Class" {:babashka/exit 1})))))
```

Reply:
Note that the Attributes class was merged to master, but hasn't been released yet. This workaround seems good though.

Reply:
@borkdude: Would you prefer to leave in the workaround for now or bump the minimum Babashka version for bbin?

Reply:
Workaround + note and then remove workaround in 6 months or so

Reply:
@jeroenvandijk: Let's update the comment here based on @borkdude's feedback. ^